src/gwkokab/inference/jenks.py

```
@@ -0,0 +1,318 @@
+        A sequence of indices representing the rightmost index of each bucket's break
+        value in the sorted data.
+    """
+    error_if(
+        n_buckets < 1 or n_buckets > len(data),
+        msg=f"Number of buckets {n_buckets} must be between 1 and {len(data)}",
+    )
+    breaks = jenkspy.jenks_breaks(data, n_classes=n_buckets)
+    n_unique_breaks = len(set(breaks))
+    if n_unique_breaks != n_buckets + 1:
+        msg = f"{n_buckets} buckets requested, but less than {n_buckets + 1} unique breaks found. "
+        ("Consider halving the number of buckets or using a different method.",)
```

Pylint (3.3.7)

[warning] Statement seems to have no effect

(W0104)